core/src/main/java/org/apache/carbondata/core/metadata/schema/table/CarbonTable.java

```diff
@@ -150,6 +150,16 @@
 
   private boolean hasDataMapSchema;
 
+  /**
+   * is local dictionary generation enabled for the table
+   */
+  private boolean isLocalDictionaryEnabled;
```

Comment:
Don't add extra properties, just get from table properties

Reply:
in stead of reading from table properties again and again and then parsing and then checking for true or false, it is better to add in carbon table, which will be build one time, and it would be easy to know whether enabled and to find default values.